Feature/connect more settings pages

tej17584 opened the pull request.

tej17584 committed:
Updating snpashots and tests

tej17584 committed:
Updating values

tej17584 committed:
Implements reset password functionality.

Adds a new screen that allows users to reset their password by submitting their email address.

The screen includes input validation and integrates with Supabase authentication to send a password reset link to the provided email.

Also, it includes I18n translations for all the text in both english and spanish.

tej17584 committed:
simulando envio

tej17584 committed:
Merge branch 'feature/settings-more-pages' into feature/connect-more-settings-pages

tej17584 committed:
Adding edit inputs for AUCTIONEER

tej17584 committed:
cambios de organzacion

tej17584 committed:
fixing typos

tej17584 committed:
helper para errores de schema

tej17584 committed:
reset password

tej17584 committed:
Testing values y actualizando schemas

tej17584 committed:
Enables edit profile functionality

Implements the edit profile feature, allowing users to update their information.

This includes:
- Integration of dynamic schema validation based on user role (User/Auctioneer).
- Supports updating user data, including handling image uploads via FormData.
- Secure API calls for updating user information.
- Displaying alerts on success or failure.
- Includes comprehensive unit tests for schema validation and payload construction for both user roles.

tej17584 committed:
Fixing testing

tej17584 committed:
format linter updates

tej17584 committed:
Adds address form modal component

Implements a modal form for creating and editing user addresses.

Includes input validation, country selection, and integration with the backend API for saving addresses. Also includes i18n support.

tej17584 committed:
Adds addresses management feature

Implements the addresses management feature, allowing users to add, view, and manage their addresses.

Introduces a new address form modal for adding new addresses.

Integrates with the user account settings to display and manage the addresses.

Removes unused constants and moves country data to the payment constants file.

tej17584 committed:
Consolidates user info edit endpoint

Uses centralized API endpoint configuration for editing user information.

This change replaces the hardcoded '/edit-user-info' string with the
`SECURE_ENDPOINTS.USER.EDIT_INFO` constant, improving maintainability
and reducing the risk of errors due to inconsistencies.

tej17584 committed:
typo error

tej17584 committed:
Improves address management UI and error handling

Enhances the address management screen with a redesigned UI, providing clearer address details and visual improvements.
Also, it adds error handling to display appropriate messages when loading addresses or updating the profile fails.

- Refactors address display for better readability.
- Improves error handling when updating user profile.
- Adds a disabled state to the "Add New" button while refreshing.

tej17584 committed:
Replaces alerts with TODO comments

Removes alert popups and replaces them with TODO comments and console errors in the edit profile and address form components.

This change is to prepare for the implementation of a new toast notification system.

tej17584 committed:
nuevo picker

tej17584 committed:
Adds addresses feature with tests

Adds address management feature, including UI components, API integration, and comprehensive testing.

The changes include:
- Implementation of an address form modal for creating new addresses.
- Implementation of an addresses screen to list user addresses.
- API config for secure address endpoints
- Mocks AsyncStorage and Picker component for tests

tej17584 committed:
Removes address form tests and adds schema tests

Removes the deprecated address form related tests to prepare for integration of new settings pages.
Adds tests for the address schema to ensure data validation.
Also installs expo-checkbox.

tej17584 committed:
Usando checkbox nativo

tej17584 committed:
Merge branch 'develop' into feature/connect-more-settings-pages

tej17584 committed:
Fixeando llamadas del hook

tej17584 committed:
snpashot update

tej17584 committed:
Fetches user data for edit profile screen

Retrieves user profile data from the API and populates the edit profile form.

This change replaces the mock user data with actual data fetched from the /user/profile endpoint. It also handles potential errors during data retrieval and redirects the user back if the data cannot be loaded. Improves user experience by pre-filling the form with current user information.

tej17584 committed:
updating snapshot

tej17584 committed:
small issues

tej17584 committed:
more copilot instructions

totisama commented on app/(tabs)/account/addresses.tsx:
```diff
@@ -1,26 +1,266 @@
+      console.error('❌ Network error loading addresses:', error);
+      // TODO: Mostrar toast genérico
+    } finally {
+      setLoading(false);
```
aquí pones loading false pero nunca lo activas no?
otra cosa, yo toda la lógica de ir a traer data la estoy metiendo dentro de un custom hook, como lo empezaste haciendo vos con lo de calendar, yo diria que sigamos con esa estructura, así si necesitamos reusarlo en algún lado (como los addresses en checkout page por ejemplo), ya tenemos el hook para solo llamarlo y obtener la data

tej17584 replied:
listo

totisama commented on app/(tabs)/account/edit-profile.tsx:
```diff
@@ -1,51 +1,256 @@
+  const userRole =
+    auth.state === 'authenticated' && auth.role ? auth.role : 'USER';
+  const schema =
+    userRole === 'AUCTIONEER' ? AuctioneerEditSchema : UserEditSchema;
```
Create un enum o una constante que solo tenga "USER" y "AUCTIONEER" como valores y saca de ahí el string y esa eso en vez de magic strings

tej17584 replied:
Okis me parece bien

totisama commented on src/hooks/api/useSecureApi.ts:
```diff
@@ -296,12 +296,25 @@ export const useSecureApi = () => {
             method: 'POST',
-            headers,
-            body: JSON.stringify(data),
+            headers: requestHeaders,
```
para qué es esto? por curiosidad jaja

tej17584 replied:
Ya te lo comente en em chat jajaja

tej17584 committed:
Enhances user role handling for profile editing

Improves type safety and maintainability by introducing constants for user roles.

This change replaces hardcoded strings with `APP_USER_ROLES` constants in the edit profile screen to determine the user's role and select the appropriate schema for form validation. This approach reduces the risk of typos and ensures consistency across the application.

tej17584 committed:
Introduces address management hooks and screen

Adds `useGetAddresses` and `useCreateAddress` hooks to handle address fetching and creation.

Implements an address screen using these hooks, providing loading, error, and empty state handling.

Refactors address form modal to use the `useCreateAddress` hook for address creation.

tej17584 committed:
Fixing addresese approach
